tests/unit/test_install.py

```diff
@@ -0,0 +1,155 @@
+
+import pytest
+from databricks.labs.blueprint.installation import Installation, MockInstallation
+from databricks.labs.blueprint.tui import MockPrompts
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import NotFound
+
+from databricks.labs.remorph.__about__ import __version__
+from databricks.labs.remorph.install import WorkspaceInstallation, WorkspaceInstaller
+
+
+@pytest.fixture
+def ws():
+    ws = create_autospec(WorkspaceClient)
+
+    ws.product = "remorph"
+    ws.product_version = __version__
+
+    return ws
+
+
+@pytest.fixture
+def mock_installation():
+    return MockInstallation(
+        {"state.json": {"version": 1, "source": "snowflake", "skip_validation": True, "catalog_name": "remorph_test"}}
+    )
+
+
+def test_install(ws, mock_installation):
+    prompts = MockPrompts(
+        {
+            r"Select the source": "0",
+            r"Do you want to Skip Validation": "yes",
+            r"Enter catalog_name": "remorph_test",
+            r"Enter schema_name": "remorph_schema",
+            r".*": "",
+        }
+    )
+    install = WorkspaceInstaller(prompts, mock_installation, ws)
+    install.run()
+
+
+def test_install_dbr(ws, mock_installation, monkeypatch):
+    monkeypatch.setenv("DATABRICKS_RUNTIME_VERSION", "14.1")
+    prompts = MockPrompts(
+        {
+            r"Select the source": "0",
+            r"Do you want to Skip Validation": "yes",
+            r".*": "",
+        }
+    )
+    with pytest.raises(SystemExit):
+        install = WorkspaceInstaller(prompts, mock_installation, ws)
+        install.run()
+
+
+def test_save_config(ws, mock_installation):
+    prompts = MockPrompts(
+        {
+            r"Select the source": "0",
+            r"Do you want to Skip Validation": "yes",
+            r"Enter catalog_name": "remorph_catalog",
+            r"Enter schema_name": "remorph_schema",
+            r"Open config file in the browser.*": "yes",
+            r".*": "",
+        }
+    )
+    install = WorkspaceInstaller(prompts, mock_installation, ws)
+    webbrowser.open = lambda x: x
+    install.configure()
+
+    mock_installation.assert_file_written(
+        "config.yml",
+        {
+            "version": 1,
+            "source": "snowflake",
+            "skip_validation": True,
+            "catalog_name": "remorph_catalog",
+            "schema_name": "remorph_schema",
+        },
+    )
+
+
+def test_create_catalog_schema(ws, mock_installation):
+    prompts = MockPrompts(
+        {
+            r"Select the source": "0",
+            r"Do you want to Skip Validation": "yes",
+            r"Enter catalog_name": "test",
+            r".*Do you want to create a new one?": True,
+            r"Enter schema_name": "schema",
+            r".*Do you want to create a new Schema?": True,
+            r".*": "",
+        }
+    )
+    install = WorkspaceInstaller(prompts, mock_installation, ws)
+    install._catalog_setup._ws.catalogs.get.side_effect = NotFound("test")
```

Comment:
Don't use private members in tests . It's prohibited for the sake of maintenance 

You already have "ws" instance - set it there